--- kernel-source/arch/arm/boot/dts/aspeed-bmc-opp-wolfpass.dts	2019-01-25 12:55:51.552977076 -0500
+++ kernel-source_mod/arch/arm/boot/dts/aspeed-bmc-opp-wolfpass.dts	2019-01-25 12:55:37.165008807 -0500
@@ -0,0 +1,309 @@
+// SPDX-License-Identifier: GPL-2.0+
+/dts-v1/;
+
+#include "aspeed-g5.dtsi"
+#include <dt-bindings/gpio/aspeed-gpio.h>
+
+/ {
+	model = "WolfPass BMC";
+	compatible = "Intel,wolfpass-bmc", "aspeed,ast2500";
+
+	chosen {
+		stdout-path = &uart5;
+		bootargs = "console=ttyS4,115200 earlyprintk";
+	};
+
+	memory@80000000 {
+		reg = <0x80000000 0x20000000>;
+	};
+
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		flash_memory: region@98000000 {
+			no-map;
+			reg = <0x98000000 0x04000000>; /* 64M */
+		};
+	};
+
+
+
+
+
+iio-hwmon {
+        compatible = "iio-hwmon";
+        io-channels = <&adc 0>, <&adc 1>, <&adc 2>, <&adc 3>,
+	<&adc 4>, <&adc 5>, <&adc 6>, <&adc 7>, <&adc 8>, <&adc 9>,
+    <&adc 10>, <&adc 11>, <&adc 12>, <&adc 13>, <&adc 14>, <&adc 15>;
+    };
+};
+
+&fmc {
+	status = "okay";
+	flash@0 {
+		status = "okay";
+		label = "bmc";
+		m25p,fast-read;
+#include "openbmc-flash-layout.dtsi"
+	};
+};
+
+&spi1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi1_default>;
+
+	flash@0 {
+		status = "okay";
+		label = "pnor";
+		m25p,fast-read;
+	};
+};
+
+&lpc_ctrl {
+	status = "okay";
+	memory-region = <&flash_memory>;
+	flash = <&spi1>;
+};
+
+&mbox {
+	status = "okay";
+};
+
+&uart1 {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_txd1_default
+			&pinctrl_rxd1_default
+			&pinctrl_nrts1_default
+			&pinctrl_ndtr1_default
+			&pinctrl_ndsr1_default
+			&pinctrl_ncts1_default
+			&pinctrl_ndcd1_default
+			&pinctrl_nri1_default>;
+};
+
+&uart2 {
+	status = "okay";
+
+       pinctrl-names = "default";
+       pinctrl-0 = <&pinctrl_txd2_default
+                       &pinctrl_rxd2_default
+                       &pinctrl_nrts2_default
+                       &pinctrl_ndtr2_default
+                       &pinctrl_ndsr2_default
+                       &pinctrl_ncts2_default
+                       &pinctrl_ndcd2_default
+                       &pinctrl_nri2_default>;
+};
+
+&uart4 {
+        status = "okay";
+};
+
+&uart5 {
+	status = "okay";
+};
+
+&mac1 {
+	status = "okay";
+	local-mac-address = [22 02 03 04 05 07];
+};
+
+&mac0 {
+	status = "okay";
+	local-mac-address = [22 02 03 04 05 06];
+	use-ncsi;
+};
+
+&espi {
+	status = "okay";
+};
+&i2c0 {
+	status = "okay";
+    #address-cells = <1>;
+    #size-cells = <0>;
+    clock-frequency = <100000>;
+};
+
+&i2c1 {
+	status = "okay";
+    #address-cells = <1>;
+    #size-cells = <0>;
+    clock-frequency = <100000>;
+    lm75@4d {
+		compatible = "ti,tmp75";
+		reg = <0x4d>;
+	};
+};
+
+&i2c2 {
+
+	status = "okay";
+	lm75@48 {
+		compatible = "ti,tmp75";
+		reg = <0x48>;
+	};
+	lm75@49 {
+		compatible = "ti,tmp75";
+		reg = <0x49>;
+	};
+};
+
+&i2c3 {
+	status = "okay";
+
+};
+
+&i2c4 {
+	status = "okay";
+
+};
+
+&i2c5 {
+	status = "okay";
+    #address-cells = <1>;
+    #size-cells = <0>;
+    clock-frequency = <100000>;
+};
+
+&i2c6 {
+	status = "okay";
+	lm75@48 {
+		compatible = "ti,tmp75";
+		reg = <0x48>;
+};
+
+	lm75@49 {
+		compatible = "ti,tmp75";
+		reg = <0x49>;
+};
+	lm75@4a {
+		compatible = "ti,tmp75";
+		reg = <0x4a>;
+};
+
+	lm75@4b {
+		compatible = "ti,tmp75";
+		reg = <0x4b>;
+};
+	lm75@4d {
+		compatible = "ti,tmp75";
+		reg = <0x4d>;
+};
+
+
+};
+
+&i2c7 {
+	status = "okay";
+
+
+};
+
+&i2c13 {
+	status = "okay";
+};
+
+
+&vuart {
+	status = "okay";
+};
+
+&gfx {
+	status = "okay";
+};
+
+&pinctrl {
+	aspeed,external-nodes = <&gfx &lhc>;
+};
+
+&wdt2 {
+	status = "okay";
+};
+
+&pwm_tacho {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm0_default
+		&pinctrl_pwm1_default
+		&pinctrl_pwm2_default
+		&pinctrl_pwm3_default
+		&pinctrl_pwm4_default
+		&pinctrl_pwm5_default>;
+	fan@0 {
+		reg = <0x00>;
+		aspeed,fan-tach-ch = /bits/ 8 <0x00>;
+	};
+	fan@1 {
+		reg = <0x00>;
+		aspeed,fan-tach-ch = /bits/ 8 <0x01>;
+	};
+	fan@2 {
+        reg = <0x01>;
+		aspeed,fan-tach-ch = /bits/ 8 <0x02>;
+	};
+	fan@3 {
+        reg = <0x01>;
+		aspeed,fan-tach-ch = /bits/ 8 <0x03>;
+	};
+	fan@4 {
+		reg = <0x02>;
+		aspeed,fan-tach-ch = /bits/ 8 <0x04>;
+	};
+	fan@5 {
+		reg = <0x02>;
+		aspeed,fan-tach-ch = /bits/ 8 <0x05>;
+	};
+	fan@6 {
+		reg = <0x03>;
+		aspeed,fan-tach-ch = /bits/ 8 <0x06>;
+	};
+	fan@7 {
+		reg = <0x03>;
+		aspeed,fan-tach-ch = /bits/ 8 <0x07>;
+	};
+	fan@8 {
+		reg = <0x04>;
+		aspeed,fan-tach-ch = /bits/ 8 <0x08>;
+	};
+	fan@9 {
+		reg = <0x04>;
+		aspeed,fan-tach-ch = /bits/ 8 <0x09>;
+	};
+	fan@10 {
+		reg = <0x05>;
+		aspeed,fan-tach-ch = /bits/ 8 <0x0a>;
+	};
+	fan@11 {
+		reg = <0x05>;
+		aspeed,fan-tach-ch = /bits/ 8 <0x0b>;
+	};
+};
+
+&adc {
+	status = "okay";
+};
+
+//&kcs1 {
+//	status = "okay";
+//	kcs_addr = <0xCA0>;
+//};
+
+//&kcs2 {
+//	status = "okay";
+//	kcs_addr = <0xCA8>;
+//};
+
+//&kcs3 {
+//	status = "okay";
+//	kcs_addr = <0xCA2>;
+//};
+
+//&lpc_snoop {
+//	status = "okay";
+//	snoop-ports = <0x80>;
+//};
